src/facets/BatchExecutionFacet.sol

```diff
@@ -0,0 +1,79 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "../storage/NiftyApesStorage.sol";
+// import "../interfaces/niftyapes/loanExecution/IBatchExecution.sol";
+import "../interfaces/niftyapes/INiftyApesStructs.sol";
+import "../interfaces/niftyapes/INiftyApesErrors.sol";
+import "../interfaces/niftyapes/INiftyApes.sol";
+import "../interfaces/seaport/ISeaport.sol";
+
+/// @title NiftyApes BatchExecution facet
+/// @custom:version 2.0
+/// @author zishansami102 (zishansami.eth)
+/// @custom:contributor captnseagraves (captnseagraves.eth)
+contract NiftyApesBatchExecutionFacet is INiftyApesStructs, INiftyApesErrors {
+    using AddressUpgradeable for address payable;
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    function buyWithSellerFinancingBatch(
+        Offer[] memory offers,
+        bytes[] calldata signatures,
+        address buyer,
+        uint256[] calldata tokenIds,
+        uint256[] calldata tokenAmounts,
+        bool partialExecution
+    ) external payable returns (uint256[] memory loanIds) {
+        // requireLengthOfAllInputArraysAreEqual
+        if (
+            offers.length != signatures.length ||
+            offers.length != tokenIds.length ||
+            offers.length != tokenAmounts.length
+        ) {
+            revert InvalidInputLength();
+        }
+
+        uint256 valueConsumed;
+        loanIds = new uint256[](offers.length);
+
+        // loop through list of offers to execute
+        for (uint256 i; i < offers.length; ++i) {
+            // instantiate calldata params to bypass stack too deep error
+            Offer memory offer = offers[i];
+            bytes calldata signature = signatures[i];
+            uint256 tokenId = tokenIds[i];
+            uint256 tokenAmount = tokenAmounts[i];
+            address buyerRe = buyer;
+
+            // try executing current offer
+            uint256 balanceBefore = address(this).balance;
+            try
+                INiftyApes(address(this)).buyWithSellerFinancing{
```

Comment:
This is the point you were making about keeping it as an external call and still being able to use try/catch? I wonder what the gas cost difference would be.

Reply:
Thinking about it, having externals calls between facets seems like a bit of an anti-pattern. It does provide a work around that allows for the use of try/catch, but it feels like a misuse of the diamond pattern and external calls.

Reply:
What if we just removed the single action functions and just made the batch functions the core functions. That would reduce the size and complexity of the system.

Reply:
It is not that uncommon in diamond patterns.
I had read an article by mudgen's regarding this, and there are alternate ways of calling functions from other facets with trade-offs.
https://eip2535diamonds.substack.com/p/how-to-share-functions-between-facets?s=w

Reply:
Also, if we merge the singe functions and batch functions in one, then the partial execution logic check has to be added at each and every revert points in the code, which itself will make the code a lot more bulkier and hard to read and maintain.
Abstracting the batch functionality with try-catch helps us simplify the core business logic, and allow us to implement the partial execution in an efficient manner.

Reply:
seize asset doesn't have partial execution logic implemented for that very reason.

Reply:
also, `msg.value` management needs to be configured for failures as well